### DISCUSSION TITLE
synced trend lines

### DISCUSSION THREAD

**Discussion by itzzp** [highfestiva/finplot#337] [Ideas]
Hello,

I would like to be able to analyze a daily and 30 minute candlestick chart and annotate the graphs. A line drawn on the daily should also be 'mirrored' on the M30 chart and vise versa. I have created a small hack: when a FinPolyLine is create on one pricechart, I create a 'mirror version' FinPolyLine on the other pricechart and the both lines get a reference to eachother. When a handle of one is moved I move the handle of the mirrored version. But this feels like a bit of a hack. Do you have plans to add support for this? Or is there a better way to solve this? Thanks in advance. 

ps I added a gif below 

![mirror_lines](https://user-images.githubusercontent.com/773814/170890183-85fb9d2a-a9ee-41b7-aed8-6a8b1945cef8.gif)

**Comment by highfestiva** [comment]
There's no plan to add support for drawing automatically in multiple axis, however I might add a nice interface to allow the user to hook onto line drawing events. Best would of course be if you supplied me with a pretty pull request?

**Reply by itzzp** [reply]
I am not into signal-slots and your code enough to be able to do this in a clean way. I was able to hack it using quite some trial and error but that's not enough for a pull request.


A short question about signals/slots. 
To store the data of a FinPolyLine I have added a line in the init of the FinPolyLine class:

self.sigRegionChangeFinished.connect(store_data_to_db) 

This triggers after the user stops dragging a handle. I noticed however that this triggers multiple times when initially drawing the line and not only once when the mouse button is released. However, when moving an existing FinPolyLine it one triggers once after releasing the mouse button. Is there a better way to trigger the function 'store_data_to_db' once after a FinPloyLine is created or changed? Thanks in advance.

**Reply by highfestiva** [reply]
I don't know unfortunately, I'm no expert on QT. GL though!
